frontend/src/components/forms/ResetPasswordForm.tsx

```diff
@@ -0,0 +1,120 @@
+import { Controller, FormProvider, useForm } from "react-hook-form";
+import { useNavigate } from "react-router-dom";
+import { Lock } from "@mui/icons-material";
+import { Stack } from "@mui/material";
+import { passwordStrength } from "check-password-strength";
+import { useSnackbar } from "notistack";
+
+import { InputWithIcon } from "src/components/InputWithIcon";
+import { StyledButton } from "src/components/StyledButton";
+import { useResetPassword } from "src/hooks/useUsers";
+import { ApiResponseError } from "src/services/ApiService";
+
+export interface ResetPasswordFormProps {
+  userId: number;
+  code: string;
+}
+
+type ResetPasswordFormState = {
+  password: string;
+  passwordConfirmation: string;
+};
+
+export const ResetPasswordForm = (props: ResetPasswordFormProps) => {
+  const { enqueueSnackbar } = useSnackbar();
+  const { resetPasswordMutation, isResetPasswordLoading } = useResetPassword();
+  const navigate = useNavigate();
+
+  const formMethods = useForm<ResetPasswordFormState>();
+
+  const { handleSubmit, getValues } = formMethods;
+
+  const onSubmit = handleSubmit(async (data: ResetPasswordFormState) => {
+    try {
+      await resetPasswordMutation({
+        userId: props.userId,
+        code: props.code,
+        password: data.password,
+      });
+      enqueueSnackbar("Successfully reset password! Please login.", {
+        variant: "success",
+      });
+      navigate("/");
+    } catch (e: unknown) {
+      enqueueSnackbar((e as ApiResponseError).message, {
+        variant: "error",
+      });
+    }
+  });
+
+  return (
+    <FormProvider {...formMethods}>
+      <Stack spacing={4} component="form" onSubmit={onSubmit} width="100%">
+        <Controller
+          name="password"
+          defaultValue={""}
+          rules={{
+            required: "Password is required.",
+            // TODO: Abstract this and improve error message.
+            validate: (value: string) => {
+              const results = passwordStrength(value);
+              return (
+                results.value === "Strong" ||
+                results.value === "Medium" ||
+                "Password is too weak."
```

Comment:
Perhaps we can let users know the password requirement: One uppercase, one lowercase and special char. Cuz it took me a while to figure out why my password did not work. But I think this can be done in another PR